/*
 * You may redistribute this program and/or modify it under the terms of
 * the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

syntax = "proto2";
import "EndpointFormat.proto";

package p2pnet.overlay;

message OverlayNode_s {
	optional bytes node_th = 1;
	optional bytes public_key = 2;

	optional string expires = 5;	// String in ISO format
	optional string lost = 6;	// Same here

	optional .p2pnet.transport.proto.TransportSocketEndpointList endpoints = 7;

	//optional bool local = 16 [default = false];
	//optional bool dht = 17 [default = false];
}